site/src/models/nfts.js

```diff
@@ -52,23 +79,59 @@ export const get = async (key) => {
 /**
  * @param {Key} key
  */
-export const remove = async (key) => stores.nfts.delete(encodeKey(key))
+export const remove = async (key) => {
+  const nft = await get(key)
+  if (!nft) return
+  await stores.nfts.delete(encodeKey(key))
+  await stores.nftsIndex.delete(
+    encodeIndexKey({ ...key, created: new Date(nft.created) })
+  )
+}
 
 /**
  * @param {any} prefix
- * @returns {Promise<NFT[]>}
+ * @param {{ limit?: number, before?: Date }} [options]
+ * @returns {Promise<Array<KVValue<NFT>>>}
  */
-export async function list(prefix) {
-  const nfts = await stores.nfts.list({
-    prefix,
-  })
-  if (nfts.keys.length > 0) {
-    return await Promise.all(
-      nfts.keys.map((key) => {
+export async function list(prefix, options) {
+  options = options || {}
+
+  const limit = Math.min(options.limit || 10, 100)
+  if (isNaN(limit) || limit < 1) {
+    throw new Error('invalid limit')
+  }
+
+  const before = options.before || new Date()
+  if (!(before instanceof Date) || isNaN(before.getTime())) {
+    throw new Error('invalid filter')
+  }
+
+  let done = false
+  let cursor
+  /** @type Promise<KVValue<NFT>>[] */
```

Comment:
why did you change the type here ?

Reply:
Since we `get` every key there's a small chance that the `get` will return null so really it's an array of `NFT|null` but we were ts-ignoring it before and it would have errored if that happened.